extensions/blocks/pinterest/pinterest.php

```diff
@@ -25,6 +29,141 @@ function register_block() {
+	$info = null;
+	if ( preg_match( URL_PATTERN, $attr['url'], $matches ) ) {
+		$info = fetch_pin_info( $matches['pin_id'] );
+	}
+
+	if ( is_array( $info ) ) {
+		$image       = $info['images']['237x'];
+		$title       = isset( $info['rich_metadata']['title'] ) ? $info['rich_metadata']['title'] : null;
+		$description = isset( $info['rich_metadata']['description'] ) ? $info['rich_metadata']['description'] : null;
+
+		// This placeholder will appear while waiting for the amp-pinterest component to initialize (or if it fails to initialize due to JS being disabled).
+		$placeholder = sprintf(
+			// The AMP_Img_Sanitizer will convert his to <amp-img> while also supplying `noscript > img` as fallback when JS is disabled.
+			'<a href="%s" placeholder><img src="%s" alt="%s" layout="fill" object-fit="contain" object-position="top left"></a>',
+			esc_url( $attr['url'] ),
+			esc_url( $image['url'] ),
+			esc_attr( $title )
+		);
+
+		$amp_padding     = 5;   // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L269>.
+		$amp_fixed_width = 237; // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L270>.
+		$pin_info_height = 60;  // Minimum Obtained by measuring the height of the .-amp-pinterest-embed-pin-text element.
+
+		// Add height based on how much description there is. There are roughly 30 characters on a line of description text.
+		$has_description = false;
+		if ( ! empty( $info['description'] ) ) {
+			$desc_padding_top = 5;  // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L342>.
+			$pin_info_height += $desc_padding_top;
+
+			// Trim whitespace on description if there is any left, use to calculate the likely rows of text.
+			$description = trim( $info['description'] );
+			if ( strlen( $description ) > 0 ) {
+				$has_description  = true;
+				$desc_line_height = 17; // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L341>.
+				$pin_info_height += ceil( strlen( $description ) / 30 ) * $desc_line_height;
+			}
+		}
+
+		if ( ! empty( $info['repin_count'] ) ) {
+			$pin_stats_height = 16;  // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L322>.
+			$pin_info_height += $pin_stats_height;
+		}
+
+		// When Pin description is empty, make sure title and description from rich metadata are supplied for accessibility and discoverability.
+		$title = $has_description ? '' : implode( ': ', array_filter( array( $title, $description ) ) );
+
+		$amp_pinterest = sprintf(
+			'<amp-pinterest style="%s" data-do="embedPin" data-url="%s" width="%d" height="%d" title="%s">%s</amp-pinterest></div>',
```

Comment:
Do you think you could switch to numbered placeholders here? I personally find it makes things easier to read when you have more than a couple of them.

Reply:
Done in 8f8012e9516eedac4cd723281c9867cbfc234c54